pkg/internal/itest/mocknet/mocknet.go

```diff
@@ -144,9 +155,11 @@ func (mcf *mockCandidateFinder) FindCandidates(ctx context.Context, cid cid.Cid)
 	for _, h := range mcf.mrn.Remotes {
 		if _, has := h.Cids[cid]; has {
 			var md metadata.Metadata
-			if h.BitswapServer != nil {
+			if h.Bitswap {
```

Comment:
might be best to use a single `Protocol` field here, it could be of type `multicodec.Cod`, with one of `multicodec.TransportBitswap`, `multicodec.TransportGraphsyncFilecoinv1`, or `multicodec.TransportIpfsGatewayHttp`. Then this can be a `switch` and you don't have the possibility of a peer serving multiple protocols—which is something we're not currently attempting to test.